src/nccl_ofi_rdma.c

```diff
@@ -3,6 +3,8 @@
  */
 #include "config.h"
 
+#define _GNU_SOURCE
```

Comment:
Given you define this in the compat path with the syscall in config bottom as well as here, better to use something like AC_GNU_SOURCE (which I just learned is obsoleted in favor of`AC_USE_SYSTEM_EXTENSIONS`)? Looks like we have a couple other places where we define it as well.

Reply:
I proposed this previously and it was contentious. So much of this initialization and setup code becomes much simpler if we're able to use libstdc++; let's leave this for now.

Reply:
I have less strong feelings about it at this point.  At the time, we still theoretically were mostly portable.  At this point, there's not even theoretically.  We're pretty well Linux only.  So _GNU_SOURCE away.

My general objection to setting _GNU_SOURCE on Linux in something like config.h is that it makes it *really* easy to accidentally use GNU extensions in code you're trying to make portable.

Reply:
OK; I think it makes sense to set `AC_USE_SYSTEM_EXTENSIONS` globally for the project then. No objection to this?

Reply:
Done